### DISCUSSION TITLE
Support light-dark() in tailwind.config with opacity modifiers

### DISCUSSION THREAD

**Discussion by simchev** [tailwindlabs/tailwindcss#15478] [Ideas]
As far as I'm aware, this isn't currently possible.

I'm trying to handle multiple dynamic themes in my app, as well as light/dark modes. My approach is to define my theme palettes in css variables and add them to tailwind.config, that way I can make them change based on both color-scheme and some custom attributes. The issue arises when I try to use the recent [light-dark()](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/light-dark) function, to avoid defining them twice with media queries. Tailwind expects variables to be simply [color channels](https://tailwindcss.com/docs/customizing-colors#using-css-variables) for them to work with the opacity modifier.

I see two options right now, which is to either define all my themes "twice" (in size, for the media query), or I can just use light-dark() directly and lose the ability to use opacity modifiers. I'll probably go with the first option, but it's a very minor annoyance. 

Thanks for looking at this 😀.

**Comment by wongjn** [comment]
You could consider using [`color-mix()`](https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/color-mix) like:

```js
module.exports = {
  theme: {
    extend: {
      colors: {
        foo: 'color-mix(in oklch, light-dark(#fff, #000) calc(<alpha-value> * 100%), transparent)',
```

**Reply by simchev** [reply]
I didn't know I had access to `<alpha-value>` there, it does the trick perfectly:

```
// my-theme.css
:root {
  --my-color: light-dark(red, orange);
}

[theme="my-theme"] {
  --my-color: light-dark(blue, purple);
}

// tailwind.config
extend: {
  colors: {
    foo: 'color-mix(in srgb, var(--my-color) calc(<alpha-value> * 100%), transparent)'
  }
}
```


Thanks a lot!
